src/io/flutter/settings/FlutterSettings.java

```diff
@@ -18,13 +18,29 @@
   private static final String reloadOnSaveKey = "io.flutter.reloadOnSave";
   private static final String openInspectorOnAppLaunchKey = "io.flutter.openInspectorOnAppLaunch";
   private static final String verboseLoggingKey = "io.flutter.verboseLogging";
-  private static final String disablePreviewDart2Key = "io.flutter.disablePreviewDart2";
+  private static final String previewDart2Key = "io.flutter.previewDart2";
   private static final String formatCodeOnSaveKey = "io.flutter.formatCodeOnSave";
   private static final String organizeImportsOnSaveKey = "io.flutter.organizeImportsOnSave";
   private static final String showOnlyWidgetsKey = "io.flutter.showOnlyWidgets";
   private static final String showPreviewAreaKey = "io.flutter.showPreviewArea";
   private static final String showHeapDisplayKey = "io.flutter.showHeapDisplay";
 
+  public enum Dart2ModeSettings {
+    useSdkDefault(0),
+    enablePreviewDart2(1),
+    disablePreviewDart2(2);
+
+    private final int ordinal;
+
+    Dart2ModeSettings(int ordinal) {
+      this.ordinal = ordinal;
```

Comment:
~Don't you get ordinals for free with Java enums?~

EDIT: OK, I think this is clearer.

Reply:
I could use, but I just want to make the order very explicit, since the UI has the same order.

Alternatively, I could pull the text from the enum itself (a user displayName field), and populate the combobox with that.

Reply:
Whatever works.  To be honest I tend to abuse enums and lean towards overly depending on things like natural order and text.  It's cute but leads to possible bugs when someone else comes along and decides to re-sort or re-name...  Color me on the fence!